### DISCUSSION TITLE
How to clear global channel map?

### DISCUSSION THREAD

**Discussion by claytonnagle** [nextgenhealthcare/connect#5102] [Q&A] [answered]
Hi all, 
I'm currently planning on adding a check to a channel (on Mirth 3.11.0) that tracks the items it processes through it via the global channel map, so items which need reprocessed can skip certain destinations. Rather than manually restarting the channel every day by hand, I want to incorporate a way to either clear the global channel map automatically each day, but the documentation does not do a good job explaining how to actually manipulate this.

Not sure if the best way to go about this is a separate channel that is timed to go off once a day to send some kind of order, or to build it into the channel itself.

Can someone help? 
Thank you

**Comment by jonbartels** [comment, marked as answer]
You can call `globalChannelMap.remove('yourkey')` to clear selected values. There is also a `clear()` method that will empty the map entirely.
